src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,572 @@
+        data.
+    resolution: float, default = 0.05
+        resolution of data to be downloaded. Can be
+        0.05 or 0.25
+
+    Examples
+    --------
+    >>> from aatoolbox import create_country_config, CodAB, ChirpsMonthly
+    >>> country_config = create_country_config(iso3="bfa")
+    >>> codab = CodAB(country_config=country_config)
+    >>> codab.download()
+    >>> admin0 = codab.load(admin_level=0)
+    >>> geo_bounding_box = GeoBoundingBox.from_shape(admin0)
+    >>> chirps_daily = ChirpsDaily(country_config=country_config,
+                                    geo_bounding_box=geo_bounding_box)
+    >>> start_date = datetime.date(year=2007, month=10, day=23)
+    >>> end_date = datetime.date(year=2020, month=3, day=2)
+
+    >>> chirps_daily.download(start_date=start_date, end_date=end_date)
+    >>> chirps_daily.process()
+    >>> chirps_daily_data = chirps_daily.load(
+    ... start_date=start_date,
+    ... end_date=end_date
+    ... )
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, frequency="daily", **kwargs)
+
+    def _get_file_name(
+        self,
+        year: Optional[str],
+        month: Optional[str],
+        day: Optional[str],
+    ) -> str:
+
+        file_name_base = self._get_file_name_base(
+            year=year,
+            month=month,
+            day=day,
+        )
+
+        file_name = (
+            f"{file_name_base}"
+            f"{day}_"
+            f"r{self._resolution}"
+            f"_{self._geobb.get_filename_repr(p=0)}.nc"
+        )
+        return file_name
+
+    def _get_last_available_date(self):
+        """Get the most recent date for which data is available."""
+        # The url contains a table where the last date available is
+        # specified.
+        url = (
+            "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/.CHIRPS/.v2p0/"
+            ".daily-improved/.global/"
+            f"{str(self._resolution).replace('.', 'p')}/"
+            ".T/last/subgrid/0./add/T/table%3A/1/%3Atable/.csv"
+        )
+
+        datetime_object = datetime.strptime(
+            pd.read_csv(url).values[0][0], "%d %b %Y"
+        )
+
+        return datetime_object.date()
+
+    def _get_url(self, year: str, month: str, day: str) -> str:
+
+        # Convert month from month number (in string format) to
+        # three-letter name
+        month_name = calendar.month_abbr[int(month)]
+
+        base_url, location_url = self._get_base_url()
+
+        url = (
+            f"{base_url}"
+            ".daily-improved/.global/."
+            f"{str(self._resolution).replace('.', 'p')}/.prcp/"
+            f"{location_url}"
+            f"T/%28{day}%20{month_name}%20{year}%29%28{day}"
+            f"%20{month_name}%20{year}"
+            "%29RANGEEDGES/data.nc"
+        )
+
+        return url
+
+    @check_file_existence
+    def _process(self, filepath: Path, ds, clobber: bool) -> Path:
```

Comment:
This method isn't covered by the tests, maybe you need to implement a test for both monthly and daily?